### ISSUE TITLE
[FEATURE] Enable Preswald to Run as a Plain Python Script (Headless Mode)

### ISSUE THREAD

**Issue by amrutha97** [StructuredLabs/preswald#541]
# Add support for running Preswald apps as standard Python scripts (no server)

---

**Overview**

Currently, Preswald requires `preswald run` to execute scripts via its FastAPI server runtime. This limits its use in environments where a full server is unnecessary or problematic (e.g. CLI workflows, cloud notebooks, automated pipelines, etc.).

We want to add a **"headless mode"** that allows users to run Preswald apps like any other Python script — from `python hello.py` — while gracefully handling or disabling UI elements and interactivity.

---

### ✅ Goals

- Execute `preswald` scripts without running `preswald run`
- Disable or noop any features that depend on the server (e.g., WebSockets, dynamic layout)
- Make components like `text`, `slider`, `plotly`, etc. fall back to CLI print or stubs
- Detect environment and auto-adapt (e.g., if run in Jupyter or `__main__`)

---

### 🛠 Tasks

- [ ] Add `ScriptService` class in `preswald.service` that mimics `PreswaldService` but runs standalone
- [ ] Add an environment detection mechanism (`PRESWALD_HEADLESS=1` or auto-detect `__main__`)
- [ ] Update core components (`text`, `alert`, `plotly`, etc.) to support headless fallback
- [ ] Add basic CLI rendering for core components (optional)
- [ ] Document usage with example:  
  ```bash
  python hello.py  # should just work!
  ```

---

### 📎 Example Usage

```python
from preswald import text, slider

value = slider("Learning Rate", default=0.01)
text(f"Training with LR: {value}")
```

Should print to the console in headless mode:
```
[slider] Learning Rate: 0.01
[text] Training with LR: 0.01
```

---

### Acceptance Criteria

- [ ] Preswald scripts run directly via `python script.py`
- [ ] No FastAPI server or frontend is started
- [ ] App does not crash due to missing socket/UI services
- [ ] Can be toggled explicitly via CLI flag or env variable

---

### Title: [FEATURE] Support TensorFlow and PyTorch usage inside Preswald apps

---

**Overview**

Many ML developers use frameworks like **TensorFlow** and **PyTorch** for model training, inference, and visualization. Currently, these workflows can break or behave unexpectedly inside Preswald due to:

- Event loops or threading from the Preswald server
- GPU resource conflicts
- Blocking UI operations

We don’t want to *wrap* or abstract these libraries, but we **do want** to make Preswald compatible with them, especially in:

- Training loops
- Metric visualizations
- Checkpointing and logging
- Real-time dashboards

---

### ✅ Goals

- Ensure TensorFlow and PyTorch code can run normally within a Preswald app
- Document usage patterns and gotchas (e.g., with `torch.cuda`, `tf.function`)
- Provide minimal example apps using both libraries
- Detect when ML libraries are being used and suggest headless mode if needed

---

### 🛠 Tasks

- [ ] Add test apps in `examples/tensorflow/` and `examples/pytorch/`
- [ ] Verify compatibility with:
  - `torch.nn`, `torch.cuda`, `torch.utils.data`
  - `tensorflow.keras`, `tf.data`, `tf.function`
- [ ] Add fallback UI methods (e.g., `plotly`, `text`, `alert`) for visualizing training progress
- [ ] Integrate `ScriptService` (from Task 1) as the default runtime for ML workflows
- [ ] Document how to run Preswald alongside TF/PT, e.g.:
  ```python
  from preswald import text, plotly

  import torch
  # training loop...

  text("Finished training.")
  ```

---

### Example Use Case

```python
# hello.py
from preswald import text, plotly
import torch.nn as nn
import torch.optim as optim

text("Starting training loop")

# Train your model
# Plot loss metrics
plotly(loss_figure)
```

**Comment by ruidazeng** [comment]
This new feature would also close #304.

**Comment by NiharP31** [comment]
Hey @amrutha97, I'd like to get assigned to this!

**Comment by shivam-singhal** [comment]
@NiharP31 feel free to put up a PR. I'll review it when it's ready :)
